python/pyspark/sql/functions.py

```diff
@@ -2633,6 +2633,23 @@ def sequence(start, stop, step=None):
             _to_java_column(start), _to_java_column(stop), _to_java_column(step)))
 
 
+@since(3.0)
+def getActiveSession():
+    """
+    Returns the active SparkSession for the current thread
+    """
+    from pyspark.sql import SparkSession
+    sc = SparkContext._active_spark_context
```

Comment:
If this is being done to simplify implementation and we don't expect people to call it directly here we should mention that in the docstring and also use an _ prefix.

I disagree with @HyukjinKwon about this behaviour being what people would expect -- it doesn't match the Scala behaviour and one of the reasons to have something like `getActiveSession()` instead of `getOrCreate()` is to allow folks to do something if we have an active session or do something else if we don't.

What about if `sc` is`None` we just return `None `since we can't have an `activeSession` without an active `SparkContext` -- does that sound reasonable?

That being said if folks feel strongly about this I'm _ok_ with us setting up a SparkContext but we need to document that if that's the path we go.

Reply:
Yea, we should match the behaviour with Scala side - that was my point essentially. The problem about the previous approach was that session was being handled within Python - I believe we will basically reuse JVM's session implementation rather than reimplementing the seperate Python session support within PySpark side.

> What about if sc isNone we just return Nonesince we can't have an activeSession without an active SparkContext -- does that sound reasonable?

In that case, I think we should follow Scala's behaviour.

Reply:
@holdenk @HyukjinKwon 
Thanks for the comments.
I checked Scala's behavior:
```
  test("my test") {
    val cx = SparkContext.getActive
    val session = SparkSession.getActiveSession
    println(cx)
    println(session)
  }
```
The result is 
```
None
None
```
So it returns None if sc isNone. Actually my current code returns None if sc isNone, but I will change the code a bit to make it more obvious. I will also add _ prefix in the function name and mention in the docstring that this function is not supposed to be called directly.